internal/pkg/deploy/sns.go

```diff
@@ -0,0 +1,89 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package deploy holds the structures to deploy infrastructure resources.
+// This file defines workload deployment resources.
+package deploy
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws/arn"
+)
+
+var errInvalidTopicARN = errors.New("ARN is not a Copilot SNS topic")
+var errInvalidARN = errors.New("invalid ARN format")
+var errInvalidARNService = errors.New("ARN is not an SNS topic")
+var errInvalidComponent = errors.New("Topic must be initialized with valid app, env, and workload")
+
+// Topic holds information about a Copilot SNS topic and its ARN, ID, and Name.
+type Topic struct {
+	awsARN arn.ARN
+	prefix string
+	wkld   string
+
+	name string
+}
+
+// NewTopic creates a new Topic struct, validating the ARN as a Copilot-managed SNS topic.
+// This function will
+func NewTopic(inputARN string, app, env, wkld string) (*Topic, error) {
```

Comment:
Can this be private?
```suggestion
func newTopic(arn, app, env, wkld string) (Topic, error) {
```

Reply:
No, because it's used in `select`.

Reply:
I can't see it in the selector pkg 👴